test/test_xpu.py

```diff
@@ -732,6 +732,38 @@ def test_get_arch_list(self):
         if not torch.xpu._is_compiled():
             self.assertEqual(len(torch.xpu.get_arch_list()), 0)
 
+@unittest.skipIf(not TEST_XPU, "XPU not available, skipping tests")
+class TestSoftmaxWithHalfToFloat(TestCase):
```

Comment:
I think it is either
- move `test_softmax_half_to_float` to `TestXpu` class, or
- rename `TestSoftmaxWithHalfToFloat` to a common name such as `TestXpuOp` and move all op-related cases such as `test_compare_cpu` and `test_non_standard_bool_values` here